File: day-2/daytwo.py

RESP_1 = 0
RESP_2 = 0

def part1(letter, passw, min, max):
    global RESP_1
    if min <= passw.count(letter) and max >= passw.count(letter):
        RESP_1 = RESP_1 + 1


def part2(letter, passw, min, max):
    global RESP_2
    if (passw[min - 1] == letter and passw[max - 1] != letter) or (passw[min - 1] != letter and passw[max - 1] == letter):
        RESP_2 = RESP_2 + 1


if __name__ == "__main__":
    with open("input.csv") as f:
        for line in f.readlines():
            letter = line.split()[1].strip(":")
            passw = line.split()[2]
            min = int(line.split()[0].split("-")[0])
            max = int(line.split()[0].split("-")[1])
        
            part1(letter, passw, min, max)
            part2(letter, passw, min, max)

    print(RESP_1)
    print(RESP_2)

File: day-2/daytwo.go
package main 

import(
	"fmt"
	"os"
	"bufio"
	"strings"
	"strconv"
)

var result_1 = 0
var result_2 = 0

func main(){
	file, _ := os.Open("input.csv") 
    scanner := bufio.NewScanner(file) 
    scanner.Split(bufio.ScanLines) 


    for scanner.Scan() {
		var line = scanner.Text() 
		var letter = strings.Trim(strings.Split(line, " ")[1], ":")
		var passw = strings.Split(line, " ")[2] 
		var min, _ = strconv.Atoi(strings.Split(strings.Split(line, " ")[0], "-")[0])
		var max,_ = strconv.Atoi(strings.Split(strings.Split(line, " ")[0], "-")[1])

		part1(letter, passw, min, max)
		part2(letter, passw, min, max)
	}

	fmt.Println(result_1)
	fmt.Println(result_2)
}

func part1(letter string, passw string, min int, max int) {
	if strings.Count(passw, letter) >= min && strings.Count(passw, letter) <= max {
		result_1++
	}
}

func part2(letter string, passw string, min int, max int) {
	if (string(passw[min - 1]) == letter && string(passw[max - 1]) != letter) || (string(passw[min - 1]) != letter && string(passw[max - 1]) == letter)  {
		result_2++
	}
}